src/main/java/com/example/eventorganizer/task/ContributionService.java

```diff
@@ -0,0 +1,31 @@
+package com.example.eventorganizer.task;
+
+import com.example.eventorganizer.event.Event;
+import com.example.eventorganizer.user.User;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.*;
+
+@Service
+public class ContributionService {
+    public Map<User, BigDecimal> calculateContributions(Event event, User spender, BigDecimal spentMoney){
```

Comment:
Good job creating separate service for this. Please add unit test to make sure is splits money correctly